src/infrastructure/validation/testing/TestHarness.java

```diff
@@ -0,0 +1,269 @@
+/**
+ * @author        Abhishek Saran, Manas Sanjay
+ * @module        Infrastructure module 
+ * @team          Test Harness
+ * @description   Test Harness main class
+ * @summary       To provide functionality to tester to run test case sets in various ways, 
+                  Methods such as runAll(), runByName(), runByCategory() and runByPriority() are provided. 
+ */
+package infrastructure.validation.testing;
+import java.io.File;
+import java.util.*;
+import infrastructure.validation.logger.LoggerFactory;
+import infrastructure.validation.logger.ILogger;
+import infrastructure.validation.logger.LogLevel;
+import infrastructure.validation.logger.ModuleID;
+
+public class TestHarness{
+  
+  private String savePath;
+  
+  public void setSavePath(String path){
+    this.savePath = path;
+  }
+  
+  public String getSavePath(){
+    return this.savePath;
+  }
+  
+  /**
+  * static helper method to return all the test cases of project
+  * @return ArrayList of File type containing all the test cases with valid naming
+  */
+  private static ArrayList<File> getAllTests(){
+    ILogger logger = LoggerFactory.getLoggerInstance();
+    ArrayList<File> allTests = new ArrayList<File>();
+    try{
+      File[] modules = new File("../../../").listFiles(File::isDirectory); 
+      for(File module : modules){
+        if(module.isDirectory()){
+          String strModule = module.getName();
+          File f = new File("src/"+strModule+"/tests/");
+          File[] tests  = f.listFiles(File::isFile);
+          for (File test : tests){
+            if(test.getName().endsWith("Test.java")){
+              allTests.add(test); 
+            }
+          } 
+        }     
+      }
+    }
+    //If any module does not content test directory or test directory is empty
+    catch (Exception e){
+      logger.log(ModuleID.TEST, LogLevel.WARNING, "Exception >> "+e);      	
+    } 
+    return allTests;
+  }
+
+
+ /**
+  * Void method to run test case of a given category
+  * @param category The name of the category of the test to be run
+  */
+  public void runByCategory(String category){
+    int totalNumberOfTests = 0;
+    int successfulNumberOfTests = 0;
+    int failedNumberOfTests = 0;
+    
+    ILogger logger = LoggerFactory.getLoggerInstance();
+    try{
+      String path = "../../../" + category + "/tests";
+
+      File directoryPath = new File(path);
+      //get all the test cases in tests directory of respective category 
+      String tests[] = directoryPath.list();
+      totalNumberOfTests = tests.length;
+      
```

Comment:
.length() Its a function right?

Reply:
No , it's a final variable, used to get the size of an array in java

Reply:
Ok